src/osa/scripts/gain_selection.py

```diff
@@ -380,13 +380,20 @@ def GainSel_finished(date: datetime) -> bool:
     """Check if gain selection finished successfully."""
     flagfile = GainSel_flag_file(date)
     return flagfile.exists()
-
+   
 
 def check_gainsel_jobs_runwise(date: datetime, run_id: int) -> bool:
     """Search for failed jobs in the log directory."""
     base_dir = Path(cfg.get("LST1", "BASE"))
     log_dir = base_dir / f"R0G/log/{date_to_dir(date)}"
     history_files = log_dir.glob(f"gain_selection_{run_id:05d}.????.history")
+    summary_table = run_summary_table(date)
```

Comment:
history_files gives you a iterator object that can be used once. Make sure that history_files iterator can be reused later when you do `for file in history_files`. Otherwise you might need to define it in the first place as a list:

```python
history_files = list(log_dir.glob(f"gain_selection_{run_id:05d}.????.history"))
```